EDU-816 Create MusicXML plugin

ahelmberger opened the pull request:
Closes https://educandu.atlassian.net/browse/EDU-816

![mxml](https://user-images.githubusercontent.com/2676617/192567083-18304f7f-1672-4c0f-9491-8c1459c04b63.png)

adrianaluchian commented on src/plugins/music-xml-viewer/music-xml-viewer.yml:
```diff
@@ -0,0 +1,9 @@
+  de: Zoom
+caption:
+  en: Caption
+  de: Legende
```
Tiny thing: the `pdf-viewer` plugin also has an identical "caption" translation, we could move it to `common.yml`

ahelmberger replied:
Done!

adrianaluchian commented on src/plugins/music-xml-viewer/music-xml-viewer-display.js:
```diff
@@ -0,0 +1,64 @@
+        <div ref={divRef} />
+      </div>
+      {!!caption && (
+        <div className={`PdfViewer-caption u-width-${width || 100}`}>
```
`MusicXmlViewer-caption`

ahelmberger replied:
Thanks for spotting this one too!

adrianaluchian commented on src/plugins/music-xml-viewer/music-xml-viewer.less:
```diff
@@ -0,0 +1,12 @@
+  margin: 0 auto;
+}
+
+.MusicXmlViewer-caption {
```
All these 3 classes could safely be called  `MusicXmlViewerDisplay[-viewer/caption]` as they are only used in the display component.

ahelmberger replied:
Done (and also done for the PdfViewer[Display], which is almost the same plugin)!

adrianaluchian replied:
Poifect 👍

adrianaluchian commented:
When I try to add a `music-xml-viewer` section in a document I get the following error:
<img width="812" alt="Screen Shot 2022-09-28 at 09 34 39" src="https://user-images.githubusercontent.com/8189923/192717381-b2256db6-6a3b-4b0e-98b2-5a43ff1fe36f.png">
Is it possible that you have some temporary local changes in this external package so that it works for you?

ahelmberger commented:
> When I try to add a `music-xml-viewer` section in a document I get the following error: <img alt="Screen Shot 2022-09-28 at 09 34 39" width="812" src="https://user-images.githubusercontent.com/8189923/192717381-b2256db6-6a3b-4b0e-98b2-5a43ff1fe36f.png"> Is it possible that you have some temporary local changes in this external package so that it works for you?

`clear` seems to throw when there was no rendering before (and I already had test sections, so it never came to my attention). Unfortunately the whole library is not very nicely documented. I will re-write his a little bit.

ahelmberger commented on src/plugins/music-xml-viewer/music-xml-viewer-info.spec.js:
```diff
@@ -0,0 +1,40 @@
+  });
+
+  describe('redactContent', () => {
+    it('redacts the PDF source url', () => {
```
Darn copy/paste, it's "XML" now :laughing:

ahelmberger commented:
Sorry for the many copy/paste errors, I was very sloppy :roll_eyes:

adrianaluchian commented:
> Sorry for the many copy/paste errors, I was very sloppy 🙄

It's the project season for such things from both of us, all good!

ahelmberger commented:
> > When I try to add a `music-xml-viewer` section in a document I get the following error: <img alt="Screen Shot 2022-09-28 at 09 34 39" width="812" src="https://user-images.githubusercontent.com/8189923/192717381-b2256db6-6a3b-4b0e-98b2-5a43ff1fe36f.png"> Is it possible that you have some temporary local changes in this external package so that it works for you?
> 
> `clear` seems to throw when there was no rendering before (and I already had test sections, so it never came to my attention). Unfortunately the whole library is not very nicely documented. I will re-write this a little bit.

I've extracted the component that renders the XML document (similar to how we've done it with PDF documents) and added some conditional logic to only clear when there was at least one render call before. Now the error is gone. Thanks for pointing it out.

adrianaluchian commented on src/components/pages/tests.js:
```diff
@@ -20,6 +20,7 @@ import MultitrackMediaEditor from '../../plugins/multitrack-media/multitrack-med
 import MultitrackMediaDisplay from '../../plugins/multitrack-media/multitrack-media-display.js';
 import { HORIZONTAL_ALIGNMENT, MEDIA_SCREEN_MODE, MEDIA_SOURCE_TYPE, STORAGE_LOCATION_TYPE, VERTICAL_ALIGNMENT } from '../../domain/constants.js';
 import { createDefaultContent, createDefaultMainTrack, createDefaultSecondaryTrack } from '../../plugins/multitrack-media/multitrack-media-utils.js';
+import MusicXmlDocument from '../music-xml-document.js';
```
These imports are staging a coup here 😆

ahelmberger replied:
:scream: I brought them to reason!

adrianaluchian commented:
> > > When I try to add a `music-xml-viewer` section in a document I get the following error: <img alt="Screen Shot 2022-09-28 at 09 34 39" width="812" src="https://user-images.githubusercontent.com/8189923/192717381-b2256db6-6a3b-4b0e-98b2-5a43ff1fe36f.png"> Is it possible that you have some temporary local changes in this external package so that it works for you?
> > 
> > 
> > `clear` seems to throw when there was no rendering before (and I already had test sections, so it never came to my attention). Unfortunately the whole library is not very nicely documented. I will re-write this a little bit.
> 
> I've extracted the component that renders the XML document (similar to how we've done it with PDF documents) and added some conditional logic to only clear when there was at least one render call before. Now the error is gone. Thanks for pointing it out.

Wonderful, looks great!! 🎉
